### PR TITLE
Use middleware but not new routers for http cors

### PR DESCRIPTION
**Pull request by lunny**
- [x] Use middlewares but not new router methods for HTTP
- [x] Added an extra route to ensue options will be requested to the middlewares.
- [x] Rename `GetOptions` / `PostOptions` back to `Get` / `Post`
- [x] Allow multiple websites on `ACCESS_CONTROL_ALLOW_ORIGIN`
- [x] Update the docs.

### CODE REVIEW

**Comment by zeripath** [comment]
You still need the OPTIONS routes in otherwise the main CORS handler will fire instead and we'll be back at the point that caused me having to add them in in the first place in #16496.

**Comment by lunny** [comment]
> You still need the OPTIONS routes in otherwise the main CORS handler will fire instead and we'll be back at the point that caused me having to add them in in the first place in #16496.

The middleware have checked the OPTIONS method. Why we still need the routes?

**Comment by zeripath** [comment]
HTTPCors will never see the options request as they will not be routed to it.

**Comment by lunny** [comment]
> HTTPCors will never see the options request as they will not be routed to it.

OK. You are right. It seems group middlewares will not be fired if no the method handler in this group. 
As a temporory resolution, I just added a placeholder options blank handler there to ensure the middlewares will be invoked.
And to resolve the problem totally, route group should be done some adjustments.

File: routers/web/web.go [hunk 1]
```diff
@@ -1032,18 +1032,19 @@ func RegisterRoutes(m *web.Route) {
...
+				m.Get("/info/refs", repo.GetInfoRefs)
+				m.Get("/HEAD", repo.GetTextFile("HEAD"))
+				m.Get("/objects/info/alternates", repo.GetTextFile("objects/info/alternates"))
+				m.Get("/objects/info/http-alternates", repo.GetTextFile("objects/info/http-alternates"))
+				m.Get("/objects/info/packs", repo.GetInfoPacks)
+				m.Get("/objects/info/{file:[^/]*}", repo.GetTextFile(""))
+				m.Get("/objects/{head:[0-9a-f]{2}}/{hash:[0-9a-f]{38}}", repo.GetLooseObject)
+				m.Get("/objects/pack/pack-{file:[0-9a-f]{40}}.pack", repo.GetPackFile)
+				m.Get("/objects/pack/pack-{file:[0-9a-f]{40}}.idx", repo.GetIdxFile)
+				m.Options("/", func(ctx *context.Context) {}) // to ensure middlewares could handle options requests
```
Comment (zeripath) [inline routers/web/web.go]: AFAIU This isn't going to fire on OPTIONS /git-upload-pack etc.

Comment (lunny) [reply routers/web/web.go]: Please review again.

**Comment by wxiaoguang** [comment]
#24303 fixed the old CORS handler.

**Pull request closed without merging**